libs/natls/src/main/java/org/amshove/natls/completion/CompletionProvider.java

```diff
@@ -71,23 +85,35 @@ public List<CompletionItem> prepareCompletion(LanguageServerFile file, Completio
 				.toList()
 		);
 
-		completionItems.addAll(localSubroutineCompletions(module));
+		completionItems.addAll(localSubroutineCompletions(module, completionContext));
 
 		completionItems.addAll(functionCompletions(file.getLibrary()));
 		completionItems.addAll(externalSubroutineCompletions(file.getLibrary()));
 		completionItems.addAll(subprogramCompletions(file.getLibrary()));
 
-		completionItems.addAll(completeSystemVars("*".equals(params.getContext().getTriggerCharacter())));
+		completionItems.addAll(completeSystemVars("*".equals(params.getContext().getTriggerCharacter()))); // TODO: Das muss wieder funktionieren
+
+		var previousTokenSource = completionContext.previousToken() != null ? completionContext.previousToken().source().toUpperCase() : null;
+		var currentTokenSource = completionContext.currentToken() != null ? completionContext.currentToken().source().toUpperCase() : null;
+		for (var completionItem : completionItems)
```

Comment:
Can this be done without looping through all items at the end? Maybe have a callback to create the data which uses the context